### PR TITLE
Feature/logs input addedit

### PR DESCRIPTION
**Pull request by radireddy**
### Requirements

* Filling out the template is required. Any pull request that does not include enough information to be reviewed in a timely manner may be closed at the maintainers' discretion.
* All new code requires tests to ensure against regressions

## Title of the Pull Requests 
Added input create/edit

### Description of the Change

1. implemented input create, wizard
2. implemented input edit, wizard
3. implemented flowgger configuration
4. implemented logstash configuration
5. implemented logstash configuration testing

### Benefits

users will be able to create input and configure logstash/flowgger

### Possible Drawbacks

none

### Applicable Issues

https://interne.ovh.net/projects/browse/UML-18

### CODE REVIEW

File: client/app/dbaas/logs/detail/inputs/add/configure/logs-inputs-add-configure.controller.js [hunk 1]
```diff
@@ -0,0 +1,399 @@
...
+
+        this.test = this.previousTest = this.ControllerHelper.request.getHashLoader({
+            loaderFunction: () => this.LogsInputsService.getTestResults(this.serviceName, this.input.data)
+        });
+
+        this.input.load()
+            .then(() => this.previousTest.load());
+    }
+
+    htmlDecode (s) {
```
Comment (Netnix) [inline client/app/dbaas/logs/detail/inputs/add/configure/logs-inputs-add-configure.controller.js]: I don't really like this function, but I understand that you need it to decode a field from the API.
Anyway, this should be moved in a service. Some comments on what it does would be welcome, if the code comes from somewhere (other than sunrise), a link to the source would be appreciated as well.

File: bower.json [hunk 2]
```diff
@@ -79,7 +79,8 @@
     "jsurl": "https://github.com/Sage/jsurl.git#v0.1.5",
     "moment-duration-format": "^1.3.0",
     "angular-websocket": "^2.0.0",
-    "ovh-angular-tail-logs": "^1.1.2"
+    "ovh-angular-tail-logs": "^1.1.2",
+    "angular-elastic": "^2.5.1"
```
Comment (Netnix) [inline bower.json]: You should use http://master.ui-kit.ovh/#!/oui-angular/textarea for your text areas. Ask @tremblaymath about improving the textarea to be resizable.

Comment (tremblaymath) [reply bower.json]: You can use ui-kit textarea. We will add it soon or you can create a pull-request, it's a guideline requirement.

File: client/app/dbaas/logs/detail/inputs/add/configure/logs-inputs-add-configure.controller.js (hunk 1 above)
Comment (radireddy) [reply client/app/dbaas/logs/detail/inputs/add/configure/logs-inputs-add-configure.controller.js]: this code is coming from sunrise. Its used to decode configuration, more like URL decoding. We will move it to service and also will explore if there is an option to get rid of it.

Comment (radireddy) [reply client/app/dbaas/logs/detail/inputs/add/configure/logs-inputs-add-configure.controller.js]: decodeUrlComponent() can be used. We are working on it

File: bower.json (hunk 2 above)
Comment (varun257) [reply bower.json]: Have fixed this in the latest commit by using a function to calculate row length. Please check the latest commit

File: client/app/dbaas/logs/detail/inputs/add/configure/logs-inputs-add-configure.controller.js (hunk 1 above)
Comment (varun257) [reply client/app/dbaas/logs/detail/inputs/add/configure/logs-inputs-add-configure.controller.js]: Have replaced data-ng-model with ng-bind-html which has dependency on ngSanitize. This actually decodes most of html entities. I dont know if it covers all cases from htmldecode function. Have to test this to know if something is missing.

File: client/app/common/translations/Messages_en_GB.xml [hunk 3]
```diff
@@ -75,6 +75,7 @@
    <translation id="edit" qtlid="37875">Edit</translation>
    <translation id="delete" qtlid="2414">Delete</translation>
    <translation id="common_back" qtlid="4118">Back</translation>
+   <translation id="common_next" qtlid="4118">Next</translation>
```
Comment (Netnix) [inline client/app/common/translations/Messages_en_GB.xml]: Remove the qtlid

File: client/app/dbaas/logs/detail/inputs/add/configure/logs-inputs-add-configure.html [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+                <label for="logs_index_modal_select" class="oui-label" data-translate="inputs_logs_configure_logstash_wizard"></label>
+                <div class="oui-field__content">
+                    <div class="oui-field-control oui-field-control_5 d-flex">
+                        <div class="oui-select">
+                            <select class="oui-select__input float-left"
+                                name="logs_inputs_configure_format"
+                                id="logs_inputs_configure_format"
+                                data-ng-model="ctrl.configuration.logstash.logFormat"
+                                data-ng-change="ctrl.applyConfiguration(ctrl.configuration.logstash.logFormat)">
+                                <option value="" data-translate="inputs_logs_configure_choose_app"></option>
```
Comment (Netnix) [inline client/app/dbaas/logs/detail/inputs/add/configure/logs-inputs-add-configure.html]: This list should not be hardcoded.

**Pull request merged**